poc_mobile_verifier/src/verifier.rs

```diff
@@ -147,14 +149,28 @@ impl Verifier {
         })
     }
 
-    pub async fn verify_epoch(
+    pub async fn verify_epoch<'a>(
         &mut self,
-        pool: impl SpeedtestStore + Copy,
-        epoch: &Range<DateTime<Utc>>,
-    ) -> Result<VerifiedEpoch> {
-        let heartbeats = Heartbeat::validate_heartbeats(&self.file_store, epoch).await?;
-        let speedtests =
-            SpeedtestAverages::validate_speedtests(pool, &self.file_store, epoch).await?;
+        pool: impl SpeedtestStore + Copy + 'a,
+        epoch: &'a Range<DateTime<Utc>>,
+    ) -> Result<
+        VerifiedEpoch<
+            impl Stream<Item = Heartbeat> + 'a,
+            impl Stream<Item = Result<SpeedtestRollingAverage>> + 'a,
```

Comment:
I like it returning a result because of the db call.  Most likely as we add more features there will be DB calls for heartbeats as well, so that will get converted to a Result.

Reply:
Is a db call any more likely to fail than any other step we already perform on heartbeat messages from ingest to reward verification?

Reply:
It's not that it's more likely, it is that if the DB call fails we want to cancel the whole transaction. We want to handle the errors differently

Reply:
tricky use of `transpose()?` there